classes/submission/maps/Schema.php

```diff
@@ -492,176 +596,212 @@ protected function getPropertyReviewRounds(Collection $reviewRounds): array
-                        'recommendOnly' => $stageAssignment->recommendOnly,
-                        'canChangeMetadata' => $stageAssignment->canChangeMetadata,
-                        'userId' => $stageAssignment->userId
-                    ];
+                // Identify data associated with editorial roles
+                if (!in_array($userGroup->roleId, [Role::ROLE_ID_MANAGER, Role::ROLE_ID_SUB_EDITOR])) {
+                    continue;
                 }
-            }
-            $stage['currentUserAssignedRoles'] = array_values(array_unique($currentUserAssignedRoles));
-            // FIXME - $stageAssignments are just temporarly added until https://github.com/pkp/pkp-lib/issues/10480 is ready
-            $stage['stageAssignments'] = $stageAssignmentsOverview;
-            if (!$stage['currentUserAssignedRoles']) {
-                if (in_array(Role::ROLE_ID_MANAGER, $currentRoles)) {
-                    $stage['currentUserAssignedRoles'][] = Role::ROLE_ID_MANAGER;
+
+                if (!$stageAssignment->recommendOnly) {
+                    $isCurrentUserDecidingEditor = true;
+                }
+
+                // if the user is assigned several times in the editorial role, and
+                // one of the assignments have recommendOnly option set, consider it here
+                if (
+                    in_array($userGroup->roleId, [Role::ROLE_ID_MANAGER, Role::ROLE_ID_SUB_EDITOR]) &&
+                    $stageAssignment->recommendOnly
+                ) {
+                    $stages[$groupStage->stageId]['currentUserCanRecommendOnly'] = true;
                 }
             }
-            // Stage-specific statuses
-            switch ($stageId) {
-                case WORKFLOW_STAGE_ID_SUBMISSION:
-                    // Replaces StageAssignmentDAO::editorAssignedToStage
-                    $assignedEditors = StageAssignment::withSubmissionIds([$submission->getId()])
-                        ->withStageIds([$stageId])
-                        ->withRoleIds([Role::ROLE_ID_MANAGER, Role::ROLE_ID_SUB_EDITOR])
-                        ->exists();
-
-                    if (!$assignedEditors) {
-                        $stage['statusId'] = Repo::submission()::STAGE_STATUS_SUBMISSION_UNASSIGNED;
-                        $stage['status'] = __('submissions.queuedUnassigned');
+        }
+
+        // if the current user is not assigned in any non-revoked role but has a global role as a manager or admin, consider it in the submission
+        if (!$isAssignedInAnyRole) {
+            $globalRoles = array_intersect([Role::ROLE_ID_MANAGER, Role::ROLE_ID_SITE_ADMIN], $this->userRoles);
+            if (!empty($globalRoles)) {
+                foreach ($stageIds as $stageId) {
+                    $stages[$stageId]['currentUserAssignedRoles'] = $globalRoles;
+                    if ($hasRecommendingEditors) {
+                        $isCurrentUserDecidingEditor = $stages[$stageId]['isCurrentUserDecidingEditor'] = true;
```

Comment:
There is logic below, to set $stages[$stageId]['isCurrentUserDecidingEditor'] based on the $isCurrentUserDecidingEditor and $hasRecommendingEditors. So maybe not necessary here. But thats very minor observation.

Reply:
I'm assigning the same value for `$isCurrentUserDecidingEditor` and `$stages[$stageId]['isCurrentUserDecidingEditor']` just for convenience for later re-use.